### PR TITLE
chore: check-pr ci job uses danger-github-oss context

### PR DESCRIPTION
**Pull request by ovasdi**
The type of this PR is: **Chore**

This PR solves [PHIRE-82](https://artsyproduct.atlassian.net/browse/PHIRE-82)

### Description

This switches the CI `check-pr` job to read _danger github token_ from CircleCI contexts. By my understanding, that is the only job that requires this var.

Also removes what seems to be a duplicate `run` instruction in that same job.

### To the Assignee 👀 

❗ Migration: ❗
- [x] unset [`DANGER_GITHUB_API_TOKEN` CircleCI project environment variable](https://app.circleci.com/settings/project/github/artsy/force/environment-variables?return-to=https%3A%2F%2Fapp.circleci.com%2Fpipelines%2Fgithub%2Fartsy%2Fforce%2F50849%2Fworkflows%2F988c696e-0ab0-49e7-8f9c-ea8e172b0e14%2Fjobs%2F381018) before merging

[GRO-434]: https://artsyproduct.atlassian.net/browse/GRO-434?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[PHIRE-82]: https://artsyproduct.atlassian.net/browse/PHIRE-82?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -161,7 +161,6 @@ jobs:
           aws_secret_access_key: $AWS_SECRET_ACCESS_KEY
     working_directory: /app
     steps:
-      - run: node scripts/strictness-migration.js check-pr
```
Comment (ovasdi) [inline .circleci/config.yml:164]: [L170](https://github.com/artsy/force/pull/12696/files#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47L170) runs the same script so removed this redundancy.

Comment (damassi) [reply .circleci/config.yml:164]: This can actually be entirely removed! We've already migrated the whole codebase.

**Review by damassi — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (ovasdi) [reply .circleci/config.yml:164]: Do you mean the whole `check-pr` job or the other `node scripts/strictness-migration.js check-pr` run ?

Comment (ovasdi) [reply .circleci/config.yml:164]: Rereading, I assume you mean the `run node scripts/strictness...`
If so, should I remove the [script](https://github.com/artsy/force/blob/main/scripts/strictness-migration.js) too?

Comment (damassi) [reply .circleci/config.yml:164]: yup and yup

Comment (ovasdi) [reply .circleci/config.yml:164]: Done in https://github.com/artsy/force/pull/12696/commits/b00850551427f3e0b1271c323654b3b3f7818343

**Review by mc-jones — APPROVED** [review]
🙏

**Pull request merged**